pkg/common/cns-lib/volume/manager.go

```diff
@@ -200,6 +208,24 @@ func ClearTaskInfoObjects() {
 				}()
 			}
 		}
+		// clear task info objects for CreateSnapshot task
+		for snapshotTaskKey, taskDetails := range snapshotTaskMap {
+			// Get the time difference between current time and the expiration
+			// time from the snapshotTaskMap.
+			diff := time.Until(taskDetails.expirationTime)
+			// Checking if the expiration time has elapsed.
+			if int(diff.Hours()) < 0 || int(diff.Minutes()) < 0 || int(diff.Seconds()) < 0 {
+				// If one of the parameters in the time object is negative, it means
+				// the entry has to be deleted.
+				log.Debugf("Found an expired taskInfo: %+v for snapshot %q. Deleting it from task map",
+					snapshotTaskMap[snapshotTaskKey].task, snapshotTaskKey)
+				func() {
+					snapshotTaskMapLock.Lock()
+					defer snapshotTaskMapLock.Unlock()
+					delete(snapshotTaskMap, snapshotTaskKey)
+				}()
```

Comment:
This looks weird, why the anonymous function? any rationale behind it?

Reply:
defer will only be triggered at the end of a func not others. Here using an anonymous function to ensure the deferred unlock is triggered timely.